homeassistant/components/unifiprotect/event.py

```diff
@@ -14,17 +12,36 @@
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
+from . import Bootstrap
 from .const import (
     ATTR_EVENT_ID,
     EVENT_TYPE_DOORBELL_RING,
     EVENT_TYPE_FINGERPRINT_IDENTIFIED,
     EVENT_TYPE_FINGERPRINT_NOT_IDENTIFIED,
     EVENT_TYPE_NFC_SCANNED,
 )
-from .data import ProtectData, ProtectDeviceType, UFPConfigEntry
+from .data import (
+    Camera,
+    EventType,
+    ProtectAdoptableDeviceModel,
+    ProtectData,
+    ProtectDeviceType,
+    UFPConfigEntry,
+)
 from .entity import EventEntityMixin, ProtectDeviceEntity, ProtectEventMixin
 
 
+def _add_ulp_user_infos(
+    bootstrap: Bootstrap, event_data: dict[str, str], ulp_id: str
+) -> None:
+    """Add ULP user information to the event data."""
+    ulp_usr = bootstrap.ulp_users.by_ulp_id(ulp_id)
+    if ulp_usr:
+        event_data["ulp_id"] = ulp_usr.ulp_id
+        event_data["full_name"] = ulp_usr.full_name
+        event_data["user_status"] = ulp_usr.status
```

Comment:
We should match the key names to https://github.com/home-assistant/core/pull/133138/files

Ideally we make them constants in that PR so we can reuse them here

Reply:
good point, done